### PR TITLE
refactor: introduce getBestQuote

### PR DESCRIPTION
**Pull request by phil-ociraptor**
# Description

Refactors the logic for getting the best quote such that one code path will work for both firm and indicative quotes. This makes the code easier to read and maintain in the future.

# Testing Instructions

Comes with unit tests.

# Checklist

-   [ ] Update [documentation](https://github.com/0xProject/website/blob/development/mdx/api/index.mdx) as needed. **Website Documentation PR:**
-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Test changes on the staging environment with [Matcha API Staging](https://api-staging.matcha.xyz).

    -   [ ] SRA/Limit orders
    -   [ ] Swap endpoints
    -   [ ] Meta transaction endpoints
    -   [ ] Depth charts

    For more information see `0x API Matcha smoke test runbook` in Quip.

### CODE REVIEW

File: src/utils/quote_comparison_utils.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+const getExpiry = (q: V4RFQIndicativeQuote | SignedNativeOrder): BigNumber => {
+    if (isSignedNativeOrder(q)) {
+        return q.order.expiry;
+    }
+
+    return q.expiry;
+};
+
+const isSignedNativeOrder = (quote: V4RFQIndicativeQuote | SignedNativeOrder): quote is SignedNativeOrder => {
```
Comment (PirosB3) [inline src/utils/quote_comparison_utils.ts:89]: Awesome! learned something new

File: src/utils/quote_comparison_utils.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+const getMakerToken = (q: V4RFQIndicativeQuote | SignedNativeOrder): string => {
+    if (isSignedNativeOrder(q)) {
+        return q.order.makerToken;
+    }
+
+    return q.makerToken;
+};
+
+const getTakerAmount = (q: V4RFQIndicativeQuote | SignedNativeOrder): BigNumber => {
```
Comment (PirosB3) [inline src/utils/quote_comparison_utils.ts:65]: could we avoid single letter variables?

**Review by PirosB3 — APPROVED** [review]
One minor comment, but thanks for teaching me something new!

File: src/utils/quote_comparison_utils.ts (hunk 2 above)
Comment (phil-ociraptor) [reply src/utils/quote_comparison_utils.ts:65]: haha yup

**Pull request merged**